opus/application/static_media/js/browse.js

```diff
@@ -796,6 +725,134 @@ var o_browse = {
         }
     },
 
+    getCurrentStartObs: function(selector, tab, viewNamespace, galleryBoundingRect, browserResized) {
+        /**
+         * get the current startObs (for slider number) and the top item's
+         * obsNum in table view, this item will also be at the top row in
+         * gallery view (for setting scrollbar location)
+         */
+
+        // this will get the top left obsNum for gallery view or the top obsNum for table view
+        let firstCachedObs = $(selector).first().data("obs");
+        let firstCachedObsTop = $(selector).first().offset().top;
+        let calculatedFirstObs = (o_utils.floor((firstCachedObs - 1)/galleryBoundingRect.x) *
+                                  galleryBoundingRect.x + 1);
+        // For gallery view, the topBoxBoundary is the top of .gallery-contents
+
+        // For table view, we will set the topBoxBoundary to be the bottom of thead
+        // (account for height of thead)
+        let topBoxBoundary = (o_browse.isGalleryView() ?
+                          $(`${tab} .gallery-contents`).offset().top :
+                          $(`${tab} .gallery-contents`).offset().top + $(`${tab} .op-data-table thead th`).outerHeight());
+
+        // table: obsNum = calculatedFirstObs + number of row
+        // gallery: obsNum = calculatedFirstObs + number of row * number of obs in a row
+        // Note: in table view, we divide by the 2nd table tr's height because in some corner
+        // cases, the first table tr's height will be 1px larger than rest of tr, and this
+        // will mess up the calculation.
+        let tableRowHeight = $(`${tab} tbody tr`).eq(1).outerHeight();
+        let obsNumDiff = (o_browse.isGalleryView() ?
+                          Math.round((topBoxBoundary - firstCachedObsTop)/o_browse.imageSize) *
+                          galleryBoundingRect.x :
+                          Math.round((topBoxBoundary - firstCachedObsTop)/
+                          tableRowHeight));
+
+        let obsNum = obsNumDiff + calculatedFirstObs;
+
+        o_browse.deleteCachedObs(browserResized, tab, viewNamespace, galleryBoundingRect,
+                                 firstCachedObs);
+
+        // Update obsNum in both infiniteScroll instances.
+        // Store the most top left obsNum in gallery for both infiniteScroll instances
+        // (this will be used to updated slider obsNum).
+        // Store the obsNum at current scrollbar location for both infiniteScroll instances
+        // (this wll be used to updated the table view scrollbar location after initInfiniteScroll
+        //  load is triggered).
+        let currentScrollObsNum = obsNum;
+        obsNum = (o_utils.floor((obsNum - 1)/galleryBoundingRect.x) *
+                  galleryBoundingRect.x + 1);
+
+        return [obsNum, currentScrollObsNum];
+    },
+
+    deleteCachedObs: function(browserResized, tab, viewNamespace ,galleryBoundingRect, firstCachedObs) {
```

Comment:
There's really no reason to pass in browserResized here, since the function does nothing if it wasn't resized. Instead, place this browserResize test back in updateSliderHandle and only call deleteCachedObs if it's true.